person.rb

```diff
@@ -0,0 +1,27 @@
+require './nameable'
+require './capitalize_decorator'
+require './trimmer_decorator'
+
+class Person < Nameable
+  attr_accessor :age, :name, :id
+
+  def initialize(age, id, name = 'unknown', parent_permission: true)
+    super()
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    true if is_of_age || @parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def correct_name
+    @name
+  end
+end
```

Comment:
Kindly include the `add_rental` (in your case `add_rental_details`) method in the `Person` class as well.  Please make sure that the person itself is sent as a parameter to create the new rental achieving the 'has-many' association.

- The method should be similar to the one you have in the `Book` class except that now you send `self` in place of `person` when creating a new rental, and `book` is received as the method parameter.